src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -224,25 +224,33 @@ WebRTCSessionStruct * WebRTCTransportProviderServer::CheckForMatchingSession(Han
     return session;
 }
 
-uint16_t WebRTCTransportProviderServer::GenerateSessionId()
+CHIP_ERROR WebRTCTransportProviderServer::GenerateSessionId(uint16_t & outSessionId)
 {
-    static uint16_t lastSessionId = 1;
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = 0;
 
-    do
+    // Try at most kMaxSessionId+1 attempts to find a free ID
+    // This ensures we never loop infinitely even if all IDs are somehow in use
+    for (uint16_t attempts = 0; attempts <= kMaxSessionId; attempts++)
     {
-        uint16_t candidateId = lastSessionId++;
+        candidateId = lastSessionId++;
 
         // Handle wrap-around per spec
         if (lastSessionId > kMaxSessionId)
         {
-            lastSessionId = 1;
+            lastSessionId = 0;
         }
 
         if (FindSession(candidateId) == nullptr)
         {
-            return candidateId;
+            outSessionId = candidateId;
+            return CHIP_NO_ERROR;
         }
-    } while (true);
+    }
+
+    // All session IDs are in use
+    ChipLogError(Zcl, "All session IDs are in use! Cannot generate new session ID.");
+    return CHIP_ERROR_MESSAGE_COUNTER_EXHAUSTED;
```

Comment:
return CHIP_IM_GLOBAL_STATUS(ResourceExhausted) and set the status based on that. Otherwise we drop the original error.